pkg/operator/resourcesynccontroller/interfaces.go

```diff
@@ -10,7 +12,15 @@ type ResourceLocation struct {
 	Provider string `json:"provider,omitempty"`
 }
 
-var emptyResourceLocation = ResourceLocation{}
+type PartialResourceLocation struct {
+	ResourceLocation
+
+	SyncedKeys sets.String `json:"syncedKeys,omitempty"`
```

Comment:
this is odd. Why does a location have the sync keys? This becomes especially strange when looking at the sync rules.

Reply:
I needed a way to store this information within the sync rules (unless you'd like to make a separate map for just the keys).